src/main/webapp/app/shared/progress-bar/student-analytics-dashboard-progress-bar.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, Input } from '@angular/core';
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
+
+@Component({
+    selector: 'jhi-student-analytics-dashboard-progress-bar',
+    templateUrl: './student-analytics-dashboard-progress-bar.component.html',
+    styleUrl: './student-analytics-dashboard-progress-bar.component.scss',
+})
+export class StudentAnalyticsDashboardProgressBarComponent {
+    @Input() size: 'small' | 'medium' | 'large' = 'medium';
+    @Input() maxValue: number = 100;
+    @Input() icon: IconDefinition | null = null;
+    @Input() progressColor: string = '#52FF00'; // Default yellow
```

Comment:
Follow coding guidelines on theming: https://docs.artemis.cit.tum.de/dev/guidelines/client-design.html#client-theming

You can use one of the already existing colors that also works for both light and dark mode.